File: templates/template.cpp
/*
 * Day ${day} - ${title}
 *
 * ${url}
 *
 *  Start: ${fetch_time}
 * Finish: TODO
 */

#include <cstdint>
#include <cstdio>
#include <iostream>
#include <ostream>
#include <string>

using std::cin;
using std::cout;
using std::endl;
using std::getline;
using std::string;


//=============== PART 1 ===============//
int64_t part1() {
    return -1; // TODO
}

//=============== PART 2 ===============//
int64_t part2() {
    return -1; // TODO
}

//=============== SOLVE ================//
int main(int argc, char *argv[]) {
    // Parse input
    for (string line; getline(cin, line); ) {
        // TODO
    }

    // Solve
    if (argc < 2) {
        std::cerr << "Please specify the part to solve." << endl;
        return 1;
    }
    switch (std::stoi(argv[1])) {
        case 1:
            cout << part1(/*TODO*/) << endl;
            break;
        case 2:
            cout << part2(/*TODO*/) << endl;
            break;
        default:
            std::cerr << "Part must be 1 or 2." << endl;
            return 1;
    }
    return 0;
}

File: templates/template.ml
(*
   Day ${day} - ${title}
   ${url}
   Start:  ${fetch_time}
   Finish: TODO
*)
(* open! Core *)

open Lib

let ( |>> ) = Utils.( |>> )

(*(*(*(*(*(*(*(*(*( PART 1 )*)*)*)*)*)*)*)*)*)
let part1 _ = raise (Invalid_argument "Part 1 not solved yet.")

(*(*(*(*(*(*(*(*(*( PART 2 )*)*)*)*)*)*)*)*)*)
let part2 _ = raise (Invalid_argument "Part 2 not solved yet.")

(*(*(*(*(*(*(*(*(*( PARSE INPUT )*)*)*)*)*)*)*)*)*)
let parsed_input =
  In_channel.input_lines In_channel.stdin |> List.map (fun line -> line)
;;

(*(*(*(*(*(*(*(*(*( SOLVE )*)*)*)*)*)*)*)*)*)
let main () =
  let solve =
    try
      match int_of_string Sys.argv.(1) with
      | 1 -> part1
      | 2 -> part2
      | _ ->
        Printf.eprintf "Part must be 1 or 2.\n";
        exit 1
    with
    | Invalid_argument _ ->
      Printf.eprintf "Please specify the part to solve.\n";
      exit 1
  in
  print_endline @@ string_of_int @@ (* Core.Tuple2.uncurry *) solve parsed_input
;;

main ()
